### ISSUE TITLE
pi07 video encoder: SpaceTimeEncoderLayerWrapper silently mixes batch items as frames when bt % num_frames == 0

### ISSUE THREAD

**Issue by shuheng-liu** [TensorAuto/OpenTau#267]
## Summary

`SpaceTimeEncoderLayerWrapper.forward` in `src/opentau/policies/pi07/low_level/video_encoder.py` infers the time axis from the **build-time** `self.num_frames`, not from the input shape. When the input's leading dim happens to be divisible by `num_frames`, the divisibility check passes and the wrapper silently reinterprets independent batch items as frames of a single video, producing incorrect output with no error.

## Where

`src/opentau/policies/pi07/low_level/video_encoder.py:284-354` — `SpaceTimeEncoderLayerWrapper.forward`:

```python
t = self.num_frames                        # build-time value (e.g. 8)
bt, n, d = hidden_states.shape
...
if t == 1:                                  # checks build-time num_frames, NOT input
    return self._spatial_block_forward(...)
if bt % t != 0:                             # only a divisibility guard
    raise ValueError(...)
b = bt // t
x = rearrange(hidden_states, "(b t) n d -> b t n d", b=b, t=t)
```

There is no check that the input *actually* has `num_frames` frames per sample. So:

| `hidden_states.shape[0]` | `num_frames` | Behavior |
| --- | --- | --- |
| `1` (B=1, single frame) | 8 | ✅ Errors out (`bt % t != 0`) |
| `8` (B=8 batch of single frames) | 8 | ❌ **Silently** rearranges into `(b=1, t=8)` and mixes batch items via temporal attention |
| `B*8` (B samples × 8 real frames) | 8 | ✅ Correct |

The outer `SpaceTimeSiglipVideoEncoder.forward` has its own guard at line 552 (`if t != self.num_frames: raise ValueError(...)`) which catches mismatches when called through the encoder. The silent path is reachable when callers go through the wrapper directly — e.g. anything that drives `vision_tower.vision_model.encoder.layers` itself, or that constructs a flattened `(B*T, N, D)` tensor with the wrong T.

## Repro

```python
import torch
from transformers import SiglipVisionConfig, SiglipVisionModel
from opentau.policies.pi07.low_level.video_encoder import SpaceTimeEncoderLayerWrapper

vt = SiglipVisionModel(SiglipVisionConfig(
    hidden_size=1152, intermediate_size=4304, num_attention_heads=16,
    num_hidden_layers=27, patch_size=14, image_size=224,
))
base_layer = vt.vision_model.encoder.layers[0]
num_patches = (224 // 14) ** 2  # 256
wrapper = SpaceTimeEncoderLayerWrapper(
    base_layer=base_layer, num_frames=8, num_tokens_per_frame=num_patches,
).eval()

# Eight independent samples, each one frame. (B=8, N=256, D=1152)
x = torch.randn(8, num_patches, 1152)
out = wrapper(x)              # no error, no warning
# Internally rearranged as (b=1, t=8); sample 0's tokens leaked into sample 7's.
```

The 8 outputs are not what you'd get from running the unwrapped `SiglipEncoderLayer` on each of the 8 samples — they have been cross-contaminated through the temporal sublayer.

## Why this matters

The MEM paper / encoder docstring promise *single-frame invariance* — a `T=1` pass should match a vanilla SigLIP forward. That guarantee depends on the caller using `suppress_spacetime_temporal(...)` (as `Gemma3WithExpertModel.embed_image` does) or on the build-time `num_frames` matching the runtime T. If a future caller forgets, the silent path produces plausible-looking outputs that are wrong.

## Suggested fix

Add an explicit shape contract on entry to `forward`. Cheapest fix: require the caller to pass T explicitly (or to declare `B`), and assert `bt == B * self.num_frames`. Failing that, at minimum reject the ambiguous case where `bt == self.num_frames` and `B` is unknowable — the wrapper has no way to distinguish "B=1 video of T frames" from "B=T single-frame samples".

Two reasonable options:

1. **Signature change**: thread `num_frames` (or `b`) through `forward` as a kwarg. The outer `SpaceTimeSiglipVideoEncoder.forward` already knows both; pass it down. Vanilla `SiglipEncoderLayer` would ignore the kwarg via the existing dispatch (`isinstance(layer, SpaceTimeEncoderLayerWrapper)` check at line 601).
2. **Default-deny**: when `_temporal_active=True`, require `bt > self.num_frames` (i.e. `B >= 2`) so the ambiguous `bt == num_frames` case is rejected. Cheaper but only patches the most common confusion; doesn't help the `bt = 2*num_frames`-but-actually-`B=2*num_frames` case.

Option 1 is the principled fix. Option 2 catches the most likely footgun.

## Acceptance

- [ ] A test that constructs `SpaceTimeEncoderLayerWrapper(num_frames=8)`, feeds a `(8, N, D)` tensor of independent samples through it, and asserts the call raises rather than silently mixing.
- [ ] Existing `tests/.../test_pi07_video_encoder_cpu.py` suite still passes.

cc @shuheng-liu

**Comment by shuheng-liu** [comment]
No longer pertinent
